### ISSUE TITLE
Sphinx 1.6.1 raising new warnings in docs build

### ISSUE THREAD

**Issue by brettcannon** [python/cpython#74565]
BPO | [30380](https://bugs.python.org/issue30380)
--- | :---
Nosy | @brettcannon, @serhiy-storchaka
PRs | <li>python/cpython#1612</li><li>python/cpython#1613</li><li>python/cpython#1614</li><li>python/cpython#1615</li><li>python/cpython#1616</li><li>python/cpython#1617</li><li>python/cpython#1618</li><li>python/cpython#1619</li><li>python/cpython#1744</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = <Date 2017-05-23.01:10:26.811>
created_at = <Date 2017-05-16.17:29:33.582>
labels = ['3.7', 'docs']
title = 'Sphinx 1.6.1 raising new warnings in docs build'
updated_at = <Date 2017-05-23.01:10:26.810>
user = 'https://github.com/brettcannon'
```

bugs.python.org fields:
```python
activity = <Date 2017-05-23.01:10:26.810>
actor = 'brett.cannon'
assignee = 'docs@python'
closed = True
closed_date = <Date 2017-05-23.01:10:26.811>
closer = 'brett.cannon'
components = ['Documentation']
creation = <Date 2017-05-16.17:29:33.582>
creator = 'brett.cannon'
dependencies = []
files = []
hgrepos = []
issue_num = 30380
keywords = []
message_count = 14.0
messages = ['293775', '293776', '293777', '293779', '293783', '293788', '293795', '293797', '293800', '293801', '293802', '293812', '293843', '294210']
nosy_count = 3.0
nosy_names = ['brett.cannon', 'docs@python', 'serhiy.storchaka']
pr_nums = ['1612', '1613', '1614', '1615', '1616', '1617', '1618', '1619', '1744']
priority = 'critical'
resolution = 'fixed'
stage = 'resolved'
status = 'closed'
superseder = None
type = None
url = 'https://bugs.python.org/issue30380'
versions = ['Python 2.7', 'Python 3.5', 'Python 3.6', 'Python 3.7']
```

</p></details>

**Comment by brettcannon** [comment]
The Travis file uses `make venv` which does `python3 -m pip install --upgrade sphinx` which is pulling in Sphinx 1.6.1 which was released today (under Python 2.7 the pip call is direct instead of using `make venv`). The problem is that Sphinx 1.6.1 adds some new warnings that then is flagging the docs build as failing.

To fix this we should probably stop calling `make venv` in .travis.yml and instead explicitly install Sphinx 1.5.6 for the docs build.

Once Travis is green again a separate PR can be done to update `master` to build cleanly under Sphinx 1.6.1.

**Comment by brettcannon** [comment]
Technically it's probably best to do `python3 -m pip install sphinx~=1.5.6` so bugfix releases are picked up.

**Comment by serhiy-storchaka** [comment]
[PR 1613](https://github.com/python/cpython/pull/1613) tries to fix a warning by using explicit footnote numbering.

**Comment by serhiy-storchaka** [comment]
New changeset d97b7dc94b19063f0589d401bdc4aaadc7030762 by Serhiy Storchaka in branch 'master':
bpo-30380: Fix Sphinx 1.6.1 warnings. (bpo-1613)
https://github.com/python/cpython/commit/d97b7dc94b19063f0589d401bdc4aaadc7030762

**Comment by brettcannon** [comment]
New changeset fca224f117d25bdfec1bf7160b67438c4fcf6dee by Brett Cannon (Mariatta) in branch 'master':
bpo-30380: Pin the version of Sphinx used to build the documentation (GH-1612)
https://github.com/python/cpython/commit/fca224f117d25bdfec1bf7160b67438c4fcf6dee

**Comment by serhiy-storchaka** [comment]
New changeset 8fc1ff51adb52c80f6bf78393413c72543bae3e1 by Serhiy Storchaka in branch '3.6':
[3.6] bpo-30380: Fix Sphinx 1.6.1 warnings. (GH-1613) (bpo-1614)
https://github.com/python/cpython/commit/8fc1ff51adb52c80f6bf78393413c72543bae3e1

**Comment by brettcannon** [comment]
New changeset 5e2016dce92d66e6e1fb25e6ce39031d8c7d5412 by Brett Cannon in branch '3.6':
bpo-30380: Pin the version of Sphinx used to build the docs (GH-1612) (GH-1617)
https://github.com/python/cpython/commit/5e2016dce92d66e6e1fb25e6ce39031d8c7d5412

**Comment by brettcannon** [comment]
New changeset 083f13e34c0b9f9ed4ed64fed3989d4af97843b3 by Brett Cannon (Serhiy Storchaka) in branch '3.5':
[3.5] bpo-30380: Fix Sphinx 1.6.1 warnings. (GH-1613) (GH-1615)
https://github.com/python/cpython/commit/083f13e34c0b9f9ed4ed64fed3989d4af97843b3

**Comment by brettcannon** [comment]
New changeset 77606957e71ce477d2c5569718f0fc36f05c6f59 by Brett Cannon in branch '3.5':
bpo-30380: Pin the version of Sphinx used to build the documentation (GH-1612) (GH-1618)
https://github.com/python/cpython/commit/77606957e71ce477d2c5569718f0fc36f05c6f59

**Comment by brettcannon** [comment]
New changeset 97eb2a7d77011605e8fc49e754cd371ead4366bc by Brett Cannon in branch '2.7':
[2.7] bpo-30380: Pin the version of Sphinx used to build the docs (GH-1612) (GH-1619)
https://github.com/python/cpython/commit/97eb2a7d77011605e8fc49e754cd371ead4366bc

**Comment by brettcannon** [comment]
All branches should be green again. Serhiy has a PR out to fix 2.7 for Sphinx 1.6.1, but it isn't passing yet so that branch is pinned to 1.5.6 until then.

I will leave this issue open for Serhiy to close once his 2.7 PR is merged.

**Comment by serhiy-storchaka** [comment]
New changeset b8b9f95f660d00ce9bd11bd9de429176858be3c5 by Serhiy Storchaka in branch '2.7':
[2.7] bpo-30380: Fix Sphinx 1.6.1 warnings. (GH-1613) (bpo-1616)
https://github.com/python/cpython/commit/b8b9f95f660d00ce9bd11bd9de429176858be3c5

**Comment by serhiy-storchaka** [comment]
There was other warnings, mostly false positive, but there were also the real documentation bugs. Sphinx now converts quotes to Unicode variants at earlier stage (this is good thing, because converting them at late stage, in HTML, not always worked), and rules in susp-ignored.csv containing '"' no longer worked.

Now warnings in 2.7 are fixed and Sphinx can be pinned to 1.6.1.

**Comment by brettcannon** [comment]
All branches are now pinned to and working under Sphinx 1.6.1, so I'm closing this as fixed!
